bootstrap/api/v1alpha1/rke2config_types.go

```diff
@@ -18,47 +18,624 @@ package v1alpha1
+	//+optional
+	ServiceNodePortRange string `json:"service-node-port-range,omitempty"`
+
+	// ClusterDNS  Cluster IP for coredns service. Should be in your service-cidr range (default: 10.43.0.10)
+	// +optional
+	ClusterDNS string `json:"clusterDNS,omitempty"`
+
+	// ClusterDomain Cluster Domain (default: "cluster.local")
+	// +optional
+	ClusterDomain string `json:"clusterDomain,omitempty"`
+
+	// WriteKubeconfig gives the path to which kubeconfig file for admin client will be written
+	// +optional
+	WriteKubeconfig string `json:"writeKubeconfig,omitempty"`
+
+	// WriteKubeconfigMode Writes kubeconfig with this mode
+	// +optional
+	WriteKubeconfigMode string `json:"writeKubeconfigMode,omitempty"`
+
+	// EtcdExposeMetrics Expose etcd metrics to client interface. (Default false)
+	// +optional
+	EtcdExposeMetrics string `json:"etcdExposeMetrics,omitempty"`
+
+	// EtcdBackupConfig defines how RKE2 will snapshot ETCD: target storage, schedule, etc.
+	//+optional
+	EtcdBackupConfig EtcdBackupConfig `json:"etcdBackupConfig,omitempty"`
+
+	// Disable Do not deploy packaged components and delete any deployed components (valid items: rke2-coredns, rke2-ingress-nginx, rke2-metrics-server)
+	// +optional
+	Disable []DisabledItem `json:"disable,omitempty"`
+
+	// DisableScheduler Disable Kubernetes default scheduler
+	// +optional
+	DisableScheduler string `json:"disable-scheduler,omitempty"`
+
+	// DisableCloudController Disable rke2 default cloud controller manager
+	// +optional
+	DisableCloudController string `json:"disableCloudController,omitempty"`
+
+	// DisableKubeProxy Disable running kube-proxy
+	// +optional
+	DisableKubeProxy string `json:"disableKubeProxy,omitempty"`
+
+	// NodeNamePrefix Prefix to be used by CAPI to generate the Node name
+	// +optional
+	NodeNamePrefix string `json:"nodeNamePrefix,omitempty"`
+
+	// NodeLabel Registering and starting kubelet with set of labels
+	// +optional
+	NodeLabels []string `json:"nodeLabel,omitempty"`
+
+	// NodeTaint Registering kubelet with set of taints
+	// +optional
+	NodeTaints []string `json:"nodeTaint,omitempty"`
+
+	// TODO: Remove ImageCredentialProviderBinDir when implementation is understood.
+	// ImageCredentialProviderBinDir The path to the directory where credential provider plugin binaries are located (default: "/var/lib/rancher/credentialprovider/bin")
+	// +optional
+	//ImageCredentialProviderBinDir string `json:"imageCredentialProviderBinDir,omitempty"`
+
+	// ImageCredentialProviderConfigMap is a reference to the ConfigMap that contains credential provider plugin config
+	// The configMap should contain a YAML file content + a Path to the Binaries for Credential Provider.
+	// +optional
+	ImageCredentialProviderConfigMap ConfigMapRef `json:"imageCredentialProviderConfigMap,omitempty"`
+
+	// TODO: Remove ContainerRuntimeEndpoint since this feature will probably not be offered by CAPI Bootstrap provider?
```

Comment:
This is Kubelet feature since Kubernetes v1.20, official documentation [here](https://kubernetes.io/docs/tasks/kubelet-credential-provider/kubelet-credential-provider/).